src/workflow/unlock.c

```diff
@@ -31,24 +32,77 @@
 
 #include <stdio.h>
 
+/**
+ * @return true if passphrase consists only of spaces, or is empty.
+ */
+static bool _is_only_spaces(const char* passphrase)
+{
+    for (const char* c = passphrase; *c != '\0'; c++) {
+        if (*c != ' ') {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * @return true if the passphrase starts or ends with a space or contains consecutive spaces.
+ */
+static bool _has_dangerous_spaces(const char* passphrase)
+{
+    size_t len = strlen(passphrase);
+    if (len == 0) {
+        return false;
+    }
+    if (passphrase[0] == ' ' || passphrase[len - 1] == ' ') {
```

Comment:
We can avoid this being dangerous if we put quotes around the passphrase when we show it (so that leading/trailing space is well visible)

Reply:
Let's postpone and make the right fix (underbrackets) sometime.